driver/registry.go

```diff
@@ -107,15 +110,28 @@ type selfServiceStrategy interface {
 }
 
 func NewRegistry(c configuration.Provider) (Registry, error) {
-	driver, err := dbal.GetDriverFor(c.DSN())
+	dsn, err := url.Parse(c.DSN())
```

Comment:
This will fail for mysql connection strings. Only parse the URL if it has prefix `sqlite://`. It's also possible that this fails for `sqlite://:memory:` which is I think a valid connection string but not a valid URL. Maybe just split at `?` and parse the query only.